doc/Doxyfile.in

```diff
@@ -58,7 +58,7 @@ PROJECT_LOGO           = doc/bitcoin_logo_doxygen.png
 # entered, it will be relative to the location where doxygen was started. If
 # left blank the current directory will be used.
 
-OUTPUT_DIRECTORY       = doc/doxygen
+OUTPUT_DIRECTORY       = @PROJECT_BINARY_DIR@/doxygen
```

Comment:
(feel free to ignore)

I find it more intuitive to put the generated documentation in the `doc` subdirectory:

```suggestion
OUTPUT_DIRECTORY       = @PROJECT_BINARY_DIR@/doc/doxygen
```

Reply:
I found a shorter path more convenient.

Not sure which approach is the optimal, though.

Reply:
Me neither. I guess that there are the following options:

1. `OUTPUT_DIRECTORY = @PROJECT_BINARY_DIR@/doxygen` (as in this PR currently, differs from `master`). Would generate the docs in `BUILD/doxygen/html/index.html` but put some supplementary files in `BUILD/doc/`.

2. `OUTPUT_DIRECTORY = @PROJECT_BINARY_DIR@/doc/doxygen` (as in `master` currently). Everything doc-related goes into `BUILD/doc/` with the generated docs at `BUILD/doc/doxygen/html/index.html`.

3. `OUTPUT_DIRECTORY = @PROJECT_BINARY_DIR@/doc`. Everything doc-related goes into `BUILD/doc/` with the generated docs at `BUILD/doc/html/index.html`.

Reply:
>     2. `OUTPUT_DIRECTORY = @PROJECT_BINARY_DIR@/doc/doxygen` (as in `master` currently). Everything doc-related goes into `BUILD/doc/` with the generated docs at `BUILD/doc/doxygen/html/index.html`.

The option 2 has been chosen during the recent call.